### DISCUSSION TITLE
✅ Automatically Generate a Typescript Client

### DISCUSSION THREAD

**Discussion by hellovai** [RobinTail/express-zod-api#321] [Ideas]
Zod is awesome! But I end up just having to manually create a shallow frontend wrapper around all the api calls I generated on the backend. 

If It was possible to automatically generate a client side library for the backend that would save a lot of time!

**Comment by RobinTail** [comment]
Have you tried this one, @hellovai ?
https://www.npmjs.com/package/@openapitools/openapi-generator-cli

**Reply by HardCoreQual** [reply]
open API scheme does not contain path parameters declarations

openapi-generator-cli error
-attribute paths.'/api/{app}/files'. Declared path parameter app needs to be defined as a path parameter in path or operation level

**Reply by RobinTail** [reply]
Actually it does:
https://github.com/RobinTail/express-zod-api/blob/master/example/example.swagger.yaml#L144
Please make sure that you declared the parameter both in routing
https://github.com/RobinTail/express-zod-api/blob/master/example/routing.ts#L15
and in the endpoint inputs
https://github.com/RobinTail/express-zod-api/blob/master/example/endpoints/update-user.ts#L11
@HardCoreQual

**Comment by RobinTail** [comment]
@hellovai @HardCoreQual ,
I'd like to inform you that the new feature — Client generation — has been released in version 6.1.0

**Comment by jokull** [comment]
I recommend [Orval](https://github.com/anymaniax/orval)!
